User registration endpoint

sealfury opened the pull request:
[API adds functionality to create an account](https://www.pivotaltracker.com/n/projects/2478342)
### User Story
```
As an API
In order to give visitors access to search functionality
I would like to be able to allow visitors to create an account
```
## Changes proposed in this pull request:
* Creates `User` model and user authentication db migration
* Writes registration and model spec tests
* Adds validation of name to `User` model
* Creates method in `application_controller` to permit name attribute
## What I have learned working on this feature:
* How to implement `devise_parameter_sanitizer`
## Packages/Gems added
* `gem 'devise_token_auth'`

sealfury committed:
Installs devise_token_auth gem
creates user model and user authentication db migration

rithreaksa committed:
Create Factorybot and a spec model for users

sealfury committed:
writes user registration spec test
adds validation of name to user model

sealfury committed:
creates method in application controller to permit name attribute, removes error message test from name attribute

emtalen commented on Gemfile:
```diff
@@ -22,6 +22,7 @@ group :development, :test do
   gem 'shoulda-matchers'
   gem 'factory_bot_rails'
   gem 'webmock'
+  gem 'devise_token_auth'
```
this needs to go in the first group, for both production, development and test

emtalen commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,11 @@
+		added_attributes = [:name]
+		devise_parameter_sanitizer.permit(:sign_up, keys: added_attributes)
+	end
 end
```
fix the indentation in this file

sealfury replied:
this appears to be indented properly already in my vscode

emtalen replied:
change from tabs to spaces 👍🏻

emtalen commented on config/initializers/devise_token_auth.rb:
```diff
@@ -0,0 +1,51 @@
+  # client is responsible for keeping track of the changing tokens. Change
+  # this to false to prevent the Authorization header from changing after
+  # each request.
+  # config.change_headers_on_each_request = true
```
you need to comment this line in

emtalen commented on spec/requests/api/user_can_register_to_search_database_spec.rb:
```diff
@@ -0,0 +1,107 @@
+      it {
+        expect(response).to have_http_status 422
+      }
+
```
remove extra line

sealfury committed:
Relocating devise_token_auth gem for prod & development
setting change headers on each request to true
cleaning up registration spec test

sealfury committed:
Removes test for invalid name from user spec
